src/mem/vmem/vmem.h

```diff
@@ -38,32 +38,44 @@ static inline vmem_page_flags_t prot_to_vmem_flags(uint32_t flags) {
 	return vmem_page_flags;
 }
 
-#define vmem_set_context(ctx)   mmu_set_context(ctx)
-
 extern int vmem_create_context(mmu_ctx_t *ctx);
-extern int vmem_init_context(mmu_ctx_t *ctx);
 extern mmu_ctx_t vmem_current_context(void);
 extern void vmem_free_context(mmu_ctx_t ctx);
 
 extern mmu_paddr_t vmem_translate(mmu_ctx_t ctx, mmu_vaddr_t virt_addr);
 
 extern int vmem_map_region(mmu_ctx_t ctx, mmu_paddr_t phy_addr, mmu_vaddr_t virt_addr,
 		size_t reg_size, vmem_page_flags_t flags);
-extern int vmem_map_region_overwrite(mmu_ctx_t ctx, mmu_paddr_t phy_addr, mmu_vaddr_t virt_addr,
-		size_t reg_size, vmem_page_flags_t flags);
+
 extern void vmem_unmap_region(mmu_ctx_t ctx, mmu_vaddr_t virt_addr, size_t reg_size);
-extern int vmem_create_space(mmu_ctx_t ctx, mmu_vaddr_t virt_addr, size_t reg_size, vmem_page_flags_t flags);
 
-extern int vmem_page_set_flags(mmu_ctx_t ctx, mmu_vaddr_t virt_addr, vmem_page_flags_t flags);
 extern int vmem_set_flags(mmu_ctx_t ctx, mmu_vaddr_t virt_addr, ssize_t len, vmem_page_flags_t flags);
 
-extern void vmem_handle_page_fault(mmu_vaddr_t virt_addr);
-
 extern void vmem_on(void);
 extern void vmem_off(void);
 extern int vmem_mmu_enabled(void);
-extern void mmu_flush_tlb(void);
 
-extern void vmem_get_idx_from_vaddr(mmu_vaddr_t virt_addr, size_t *pgd_idx, size_t *pmd_idx, size_t *pte_idx);
+#define MMU_LAST_LEVEL (MMU_LEVELS - 1)
+
+#ifndef __MMU_SHIFT_1
+#define __MMU_SHIFT_1 (0 * 8 * sizeof(uintptr_t))
```

Comment:
Why not just simply `#define __MMU_SHIFT_1 0`?

Reply:
done